### ISSUE TITLE
Retry or exit when failing to start with pulseaudio backend

### ISSUE THREAD

**Issue by tmigone** [mikebrady/shairport-sync#1054]
**Scenario**

- Start `shairport-sync` with a `pulseaudio` backend (for example using the following command `shairport-sync --name "test" --output pa`)
- Purposely **stop** my `pulseaudio` server so that `shairport-sync` can't connect to `pulseaudio`.
- I get the following error: `fatal error: pa context is not good -- the error message "Connection refused".`

**Problem**

It's fine that I get the error but the program does not exit nor does it retry to connect to `pulseaudio` after a while. I'm running `shairport-sync` and `pulseaudio` in docker containers so this means that if my `pulseaudio` server is not ready by the time `shairport-sync` starts it won't ever connect.

**Proposed solutions**

1. Make `shairport-sync` retry the connection to `pulseaudio` after a few seconds.
2. Make `shairport-sync` exit when the error above gets thrown.

Option 1 is what I've seen most programs similar to `shairport-sync` do.

**Comment by mikebrady** [comment]
Thanks for the report. Proposed solution (2) makes the most sense to me right now. I'll take a look at it.

**Comment by tmigone** [comment]
Much appreciated!

**Comment by mikebrady** [comment]
Hi there. Could you tell me how you are stopping the `pulseaudio` server please? I'm running Ubuntu 2004 and when I kill the process it just respawns, I guess. Shairport Sync doesn't even notice it's gone, which isn't good.

Also, what version of Shairport Sync?

**Comment by tmigone** [comment]
Hi, I'm using docker to run pulseaudio so I just don't start the container. For Ubuntu, I guess one of the following should work, not sure which one.

```
systemctl --user stop pulseaudio.socket
systemctl --user stop pulseaudio.service
pulseaudio -k
```

Another trick you can try is to tell `shairport-sync` to connect to a different (non-existant) pulseaudio server with an env var, this should have the same effect and maybe it's easier.:

```
PULSE_SERVER=<whatever-garbage> shairport-sync --name "test
```


I'm using the latest available as a debian package (3.2.2)

**Comment by mikebrady** [comment]
Thanks. So, I had a good look at this. 

The latest version of Shairport Sync does exit when it receives that error. The current version is 3.3.7rc1 and so 3.2.2 is pretty ancient, and there have been many fixes to it in the meantime, including exiting cleanly.

There is a docker version on the Docker Hub at https://hub.docker.com/repository/docker/mikebrady/shairport-sync.
